### PR TITLE
Validate pencil halo and padding extents

### PR DESCRIPTION
**Pull request by fallintoplace**
## Summary
- Reject negative halo extents and padding in `cudecompGetPencilInfo`.
- Check computed pencil shapes before assigning them to the `int32_t` shape fields while preserving valid zero-sized empty pencils.
- Reject pencil size multiplication that would overflow `int64_t`, including zero-size handling.
- Add API coverage for invalid extents, empty pencils, oversized shapes, and size overflow.

## Validation
- `git diff --check`

### CODE REVIEW

File: src/cudecomp.cc [hunk 1]
```diff
@@ -272,6 +273,30 @@ static void checkConfig(cudecompHandle_t handle, const cudecompGridDescConfig_t*
...
+static int32_t getNonnegativePencilExtent(const int32_t extents[], int dim, const char* name) {
+  int32_t value = extents ? extents[dim] : 0;
+  if (value < 0) {
+    std::string message = std::string(name) + " values must be non-negative";
+    THROW_INVALID_USAGE(message.c_str());
+  }
+  return value;
+}
+
+static int32_t checkedPencilShape(int64_t shape) {
```
Comment (romerojosh) [inline src/cudecomp.cc:285]: We currently support users creating grid descriptors/pencil info structures with empty pencils (see #103 for more context). As a result, `checkedPencilShape` rejecting `shape <= 0` is too strong. Empty local pencils can legitimately have a zero shape, and `cudecompGetPencilInfo` should still describe them. This check should allow zero and only reject negative or `int32_t` overflow values. If zero is allowed, `checkedPencilSizeProduct` also needs to handle `size == 0` without dividing by zero.

**Review by romerojosh — COMMENTED** [review]
Hi @fallintoplace, thanks for the contribution. Left a comment for you to address with this one.

File: tests/ctest/api_tests.cc [hunk 2]
```diff
@@ -1050,15 +1051,60 @@ TEST_F(ApiGetPencilInfoTest, MatchesExpectedGdimsDistDecomposition) {
   }
 }
 
+TEST_F(ApiGetPencilInfoTest, DescribesEmptyPencils) {
```
Comment (romerojosh) [inline tests/ctest/api_tests.cc:1054]: Thanks for adding this test to enforce the current empty pencil support!

**Review by romerojosh — APPROVED** [review]
LGTM! Will merge once build and tests pass. Thanks for the contribution.
